aea/configurations/loader.py

```diff
@@ -308,18 +308,20 @@ def from_package_type(
 
 
 def load_component_configuration(
-    component_type: ComponentType,
+    component_type: Union[ComponentType, PackageType],
     directory: Path,
     skip_consistency_check: bool = False,
 ) -> "ComponentConfiguration":
```

Comment:
This is not ok imo. PackageType includes Agent, ComponentType does not. This is confusing terminology. You can create a new method which handles both, but this method should stay as is and be consistently named as is.